### PR TITLE
Lock down node.js to version 12 for sample apps

### PR DESCRIPTION
**Pull request by katsutoxin**
connect to #251 

@viktorklang met the problem to install js-shopping-cart in node version 14.  And it works if switching back to node version 12.

I think that we should lock down node to specific version in our sample app.
Currently it is a little bit messed up. I found that in the current master branch
-  in Dockerfile, we use node version 8
- in Circle Build, we use node version 13

I will move everything in this repo to node version 12, and introduce nvm(node version manager) to control the node version, nvm will read `.nvmrc` file to pick up correct node version with command `nvm install && nvm use`

If the node version is incorrect, when user do `npm install`, user will see the error and stop the build (Note: it is put in engine field in package.json and .npmrc file)
```
18:07 $ npm install
npm ERR! code ENOTSUP
npm ERR! notsup Unsupported engine for cloudstate-shopping@1.0.0: wanted: {"node":"~12"} (current: {"node":"14.2.0","npm":"6.14.4"})
npm ERR! notsup Not compatible with your version of node/npm: cloudstate-shopping@1.0.0
npm ERR! notsup Not compatible with your version of node/npm: cloudstate-shopping@1.0.0
npm ERR! notsup Required: {"node":"~12"}
npm ERR! notsup Actual:   {"npm":"6.14.4","node":"14.2.0"}

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/danny/.npm/_logs/2020-05-18T01_07_56_869Z-debug.log
```

### CODE REVIEW

**Comment by katsutoxin** [comment]
> Should there be any notes or comments about why not using version 13 today? Or notes about future upgrade considerations to otherwise be kind to future maintainers?

Good point. The reason that I picked up node version 12 is based on node.js website https://nodejs.org/en/    Currently version 12 is recommended for most users. 

More detail is at the release website https://nodejs.org/en/about/releases/   - After six months, odd-numbered releases (9, 11, etc.) become unsupported, and even-numbered releases (10, 12, etc.) move to Active LTS(Long Term Support) status and are ready for general use.  

Based on the release website, version 13 will be end of life in 6/1/2020, which is around two weeks from today. It is why I didn't use this version.

**Comment by edwardcallahan** [comment]
Thanks!

File: chat/chat/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:8.15
+FROM node:12.16.3
```
Comment (vkorenev) [inline chat/chat/Dockerfile:1]: How's about using `node:12.16.3-buster-slim` instead? Its size is 154MB while the size of `node:12.16.3` is 916MB.

Comment (katsutoxin) [reply chat/chat/Dockerfile:1]: @vkorenev  Good suggestion.  I need to test on it.  BTW, `node:12.16.3` is   335.39 MB , `12.16.3-buster-slim` is 51.54MB  if OS/ARCH is linux/amd64. Another option is to use `12.16.3-alpine`, it is only 28.64MB
   https://hub.docker.com/_/node/?tab=tags&page=1&name=12.16.3

Comment (viktorklang) [reply chat/chat/Dockerfile:1]: @katsutoxin I think there are some security vulnerabilities which makes buster preferred over alpine, but I think @srikanthops may know more.

Comment (vkorenev) [reply chat/chat/Dockerfile:1]: @katsutoxin you are right. I looked on unpacked image size.

**Comment by srikanthops** [comment]
@katsutoxin 
Did you take a look at this
https://github.com/lightbend/cloudstate-samples/network/alerts

File: chat/chat/Dockerfile (hunk 1 above)
Comment (srikanthops) [reply chat/chat/Dockerfile]: @katsutoxin I would go with `12.16.3-buster-slim` buster has some vulnerabilities addressed..

**Comment by katsutoxin** [comment]
> https://github.com/lightbend/cloudstate-samples/network/alerts

@srikanthops what is it?  My browser shows "404 not found"

**Comment by srikanthops** [comment]
it s fix for this vulnerability  @katsutoxin 
https://github.com/advisories/GHSA-vh95-rmgr-6w4m

code changes should look like
```
  "dependencies": {
  "minimist": ">=0.2.1"
}
```

**Comment by katsutoxin** [comment]
> "minimist": ">=0.2.1"

Hi @srikanthops ,  This is a known issue happened last month https://github.com/grpc/grpc-node/issues/1363

We need to wait for grpc-node library to fix it.

File: chat/chat/Dockerfile (hunk 1 above)
Comment (katsutoxin) [reply chat/chat/Dockerfile]: sounds good.  I made the code change to use 12.16.3-buster-slim

**Pull request merged**